### PR TITLE
Chapter 6

### PR DESCRIPTION
**Pull request by bast**
Updating the book text as I go. Work in progress.

### CODE REVIEW

**Comment by bast** [comment]
Open issues:
- [ ] 6/2 is not updated in the book - first want to get some feedback. Rest is in sync.

File: chapter-06/recipe-02/fortran-c-example/CMakeLists.txt [hunk 1]
```diff
@@ -29,25 +29,31 @@ cmake_host_system_information(RESULT _os_platform QUERY OS_PLATFORM)
...
-            os_platform='${_os_platform}',
-            cmake_version='${CMAKE_VERSION}',
-            cmake_generator='${CMAKE_GENERATOR}',
-            fc='${CMAKE_Fortran_COMPILER}',
-            cc='${CMAKE_C_COMPILER}')
-cf.configure_file(conf_dict, 'print_info.c', in_path='${CMAKE_CURRENT_SOURCE_DIR}', out_path='${CMAKE_CURRENT_BINARY_DIR}', suffix='.in')
+import sys
+sys.path.insert(0, '${CMAKE_CURRENT_SOURCE_DIR}')
+
+import os
```
Comment (robertodr) [inline chapter-06/recipe-02/fortran-c-example/CMakeLists.txt]: Can we use `pathlib` here and in the file operations?

File: chapter-06/recipe-02/fortran-c-example/configurator.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+def configure_file(input_file, output_file, vars_dict):
```
Comment (robertodr) [inline chapter-06/recipe-02/fortran-c-example/configurator.py]: I like the fact that preparing the configuration dictionary has been factored out. Indeed it makes slightly more sense like this in the original context for which this recipe was thought.

File: chapter-06/recipe-02/fortran-c-example/CMakeLists.txt (hunk 1 above)
Comment (bast) [reply chapter-06/recipe-02/fortran-c-example/CMakeLists.txt]: OK I will try with pathlib.

**Comment by bast** [comment]
This is now ready to go although currently without Appveyor testing (I am working on re-enabling that).

**Comment by bast** [comment]
Appveyor is back!

**Comment by bast** [comment]
Before merging we need to check #397.

**Comment by bast** [comment]
As soon as #399 is in, I will rebase this PR and clean up test problems.

**Comment by bast** [comment]
This will probably green up the tests but before merging I need to adjust the recipe on TC as well as fix the `fqdn` typo.

**Comment by bast** [comment]
- [ ] fqdn 6/1
- [ ] fqdn 6/2
- [ ] simplify 6/2

**Pull request merged**